### ISSUE TITLE
Unable to retrieve attribute from context node

### ISSUE THREAD

**Issue by Jacks50** [goto100/xpath#65]
I got a little problem when trying to retrieve an attribute from a contextual node. Here is what I've done :

**Retrieved the node from an xml :** 
[Here](http://paste.ubuntu.com/23593930/) is what my node looks like with `console.log(node + "\n ----")`. The node is retrieved from a `Document` using `xpath.select("//Regimens/Regimen[1]", xmlDoc)`.

(I had to put the `\n ----` in order to have a string output, maybe this is an hint ?).

What I'm trying to retrieve is the "Drug" attribute of the node.

**Here is what I do in my code :**
`return xpath.select1("//@Drug", node).value;`

But it gives me the following error : `Cannot read property 'value' from undefined`, so it seems it doesn't find the attribute, but why ?

I'm using the last version of `xpath` with TypeScript. Retrieving nodes works well, but I stumbled on this problem and can't figure out what's wrong. Thank you very much.

**Comment by JLRishe** [comment]
@Jacks50  Please show us the actual code you're trying to use instead of small pieces of it. Selecting an attribute from a context node should definitely work.

**Comment by Jacks50** [comment]
Hi @JLRishe , thank you for answering. It seems it was a problem in my code indeed. I was retrieving single nodes by using the `xpath.select(query, doc)` which sets the `single` parameter to false if unspecified. Then, I used these single nodes as a context to retrieve attributes. Could that lead to an error ?

I resolved my issue when writing a Test class to provide you, and when rewriting the code I noticed this, which made me able to retrieve the attribute I was looking for.
